src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/SafeCryptoHandles.cs

```diff
@@ -182,9 +178,34 @@ internal bool PublicOnly
             }
         }
 
+        internal void SetParent(SafeProvHandle parent)
+        {
+            if (IsInvalid || IsClosed)
+            {
+                return;
+            }
+
+            Debug.Assert(_parent == null);
+            Debug.Assert(!parent.IsClosed);
+            Debug.Assert(!parent.IsInvalid);
+
+            _parent = parent;
+
+            bool ignored = false;
+            parent.DangerousAddRef(ref ignored);
```

Comment:
Ah, I see, that the pending fieldref would forcibly avoid the situation where the object got collected.  That's what I missed. ("But... that's obvious", well, perhaps, but since `this` getting finalized while an instance method is executing is already non-obvious enough that I was suspending all prejudice)

The exception I was referring to is DangerousAddRef on an IsClosed handle:  https://github.com/dotnet/coreclr/blob/43b39a73cbf832ec13ec29ed356cb75834e7a8d7/src/vm/safehandle.cpp#L112

But that doesn't apply, since the field ref stopped the finalize from happening, which stopped the release from happening.